src/main/java/io/castle/client/internal/backend/OkRestApiBackend.java

```diff
@@ -0,0 +1,166 @@
+        this.track = baseUrl.resolve("/v1/track.json");
+        this.authenticate = baseUrl.resolve("/v1/authenticate");
+    }
+
+    @Override
+    public void sendTrackRequest(String event, String userId, JsonElement contextPayload, JsonElement propertiesPayload) {
+        sendTrackRequest(event, userId, contextPayload, propertiesPayload, null);
+    }
+
+    @Override
+    public void sendTrackRequest(String event, String userId, JsonElement contextPayload, JsonElement propertiesPayload, final AsyncCallbackHandler<Boolean> asyncCallbackHandler) {
+        JsonObject json = new JsonObject();
+        json.add("name", new JsonPrimitive(event));
+        if (userId == null) {
+            json.add("user_id", JsonNull.INSTANCE);
+        } else {
+            json.add("user_id", new JsonPrimitive(userId));
+        }
+        json.add("context", contextPayload);
+        if (propertiesPayload != null) {
+            json.add("properties", propertiesPayload);
+        }
+        String content = json.toString();
+        RequestBody body = RequestBody.create(JSON, content);
+        Request request = new Request.Builder()
+                .url(track)
+                .post(body)
+                .build();
+        client.newCall(request).enqueue(new Callback() {
+            @Override
+            public void onFailure(Call call, IOException e) {
+                Castle.logger.error("HTTP layer. Error sending track request.", e);
+                if (asyncCallbackHandler != null) {
+                    asyncCallbackHandler.onResponse(false);
+                }
+            }
+
+            @Override
+            public void onResponse(Call call, Response response) throws IOException {
+                if (asyncCallbackHandler != null) {
+                    asyncCallbackHandler.onResponse(response.isSuccessful());
+                }
+            }
+        });
+    }
+
+    private Request buildAuthenticateRequest(String event, String userId, JsonElement contextPayload, JsonElement propertiesPayload) {
+        JsonObject json = new JsonObject();
+        json.add("name", new JsonPrimitive(event));
+        json.add("user_id", new JsonPrimitive(userId));
+        json.add("context", contextPayload);
+        if (propertiesPayload != null) {
+            json.add("properties", propertiesPayload);
+        }
+        RequestBody body = RequestBody.create(JSON, json.toString());
+        return new Request.Builder()
+                .url(authenticate)
+                .post(body)
+                .build();
+    }
+
+    @Override
+    public AuthenticateAction sendAuthenticateSync(String event, String userId, JsonElement contextPayload, JsonElement propertiesPayload) {
+        Request request = buildAuthenticateRequest(event, userId, contextPayload, propertiesPayload);
+        try {
+            Response response = client.newCall(request).execute();
+            return extractAuthenticationAction(response);
```

Comment:
regarding fallback strategies see https://paper.dropbox.com/doc/Server-SDK-specs-5WNJdpiGefPk2I1aAtWET#:h2=Fail-over-Strategy: 
please update behavior accordingly

Reply:
Yes, the throw option is not implemented. To be fixed.